index.html

```diff
@@ -218,6 +237,400 @@ <h2>
+            Home tab invariants
+          </h3>
+          <p>
+            The above rules are intended to ensure that the following
+            invariants are always true, for applications that [=has a home
+            tab|have a home tab=]:
+          </p>
+          <ul>
+            <li>every application window has exactly one [=home tab context=],
+            and
+            </li>
+            <li>every [=home tab context=]'s active document's [=Document/URL=]
+            is [=within home tab scope=] (unless the document's URL has changed
+            since it was created), and
+            </li>
+            <li>every non-home-tab [=application context=]'s active document's
+            [=Document/URL=] is [=outside of home tab scope=] (unless the
+            document's URL has changed since it was created).
+            </li>
+          </ul>
+          <p>
+            User agents will not dynamically move documents between home-tab
+            and non-home-tab contexts if they change their [=URL/fragment=], or
+            use the {{History}} API to modify their display URL into or out of
+            the home tab scope, because no navigation is taking place. For this
+            reason, the above invariants only care about the [=Document/URLs=]
+            that documents had at the time of their creation.
+          </p>
+          <p>
+            Applications that "pretend" to navigate by modifing their URLs can
+            detect when they are in tabbed application mode and change their
+            behaviour to perform actual navigations into and out of the home
+            tab scope, rather than modifying the URL, to ensure the user agent
+            correctly opens documents in or out of the [=home tab context=].
+          </p>
+        </section>
+      </section>
+      <section>
+        <h3>
+          <code><dfn>new_tab_button</dfn></code> member
+        </h3>
+        <p>
+          The [=tab_strip/new_tab_button=] member is an ordered map that
+          describes the behaviour of a UI affordance (such as a button) which,
+          when clicked/activated, opens a new [=application context=] within
+          the application window. It has the following members:
+        </p>
+        <ul>
+          <li>[=new_tab_button/url=]
+          </li>
+        </ul>
+        <p>
+          The <code><dfn data-dfn-for="new_tab_button">url</dfn></code> member
+          is a string that represents a URL relative to the [=manifest URL=]
+          that is [=manifest/within scope=] of a [=Document/processed
+          manifest=].
+        </p>
+        <aside class="note">
+          The [=new_tab_button=] member will be ignored if it is [=within home
+          tab scope=].
+        </aside>
+        <p>
+          When the new tab button is activated by the end user, the following
+          steps are run:
+        </p>
+        <ol>
+          <li>Create a new [=application context=] in the current window and
```

Comment:
perhaps reference the HTML spec algo too somehow? https://html.spec.whatwg.org/multipage/document-sequences.html#creating-a-new-browsing-context

Reply:
Actually there is an existing definition of [create a new application context](https://www.w3.org/TR/appmanifest/#dfn-create-a-new-application-context) which I can leverage here.